index.html

```diff
@@ -561,21 +571,34 @@ <h3>
         <ol class="algorithm">
           <li>Let |lock| be the <a>context object</a>.
           </li>
-          <li>Let |record| be the <a>platform wake lock</a>'s <a>state
-          record</a> associated with |lock|'s |type|.
-          </li>
-          <li>If the promise |lock|.<a>[[\requestPromise]]</a> is not
-          `undefined`, return it.
-          </li>
           <li>Let |promise:Promise| be <a data-cite=
           "promises-guide#a-new-promise">a new promise</a>.
           </li>
+          <li>If the promise |lock|.<a>[[\requestPromise]]</a> is not
```

Comment:
If there is already a `[[requestPromise]]` that has not been resolved shouldn't we attach the "release a wake lock" steps to `signal` and return a existing promise.

Reply:
Yes but I think that if I did that they the above "signal" replacement idea wouldn't work - I would have to look and maybe adopt the algorithm

Reply:
It's not a replacement, the new signal provides an additional way to initiate the abort steps.

Reply:
True, right... just need to execute the steps in the right order then